### PR TITLE
[bazel] Convert boot_policy_{newer, rollback} to the new rules

### PR DESCRIPTION
**Pull request by cfrantz**
Fixes: #20086
Fixes: #20103

### CODE REVIEW

File: sw/device/silicon_creator/rom/e2e/boot_policy_newer/BUILD [hunk 1]
```diff
@@ -121,42 +102,35 @@ BOOT_POLICY_NEWER_CASES = [
...
-    opentitan_functest(
+    opentitan_test(
         name = "boot_policy_newer_{}_a_{}_b_{}".format(
             lc_state,
             t["a"],
             t["b"],
         ),
         cw310 = cw310_params(
-            bitstream = ":bitstream_boot_policy_newer_{}".format(lc_state),
+            assemble = "{fw_a}@0 {fw_b}@0x80000",
```
Comment (pamaury) [inline sw/device/silicon_creator/rom/e2e/boot_policy_newer/BUILD:122]: maybe could use `SLOTS` in `//sw/device/silicon_creator/rom/e2e:defs.bzl`

File: sw/device/silicon_creator/rom/e2e/boot_policy_rollback/BUILD [hunk 2]
```diff
@@ -71,44 +67,36 @@ otp_json(
...
-    opentitan_functest(
+    opentitan_test(
         name = "boot_policy_rollback_{}_a_{}_b_{}".format(
             lc_state,
             t["a"],
             t["b"],
         ),
         cw310 = cw310_params(
-            bitstream = "bitstream_boot_policy_rollback_{}".format(lc_state),
+            assemble = "{fw_a}@0 {fw_b}@0x80000",
```
Comment (pamaury) [inline sw/device/silicon_creator/rom/e2e/boot_policy_rollback/BUILD:88]: Same.

**Review by pamaury — APPROVED** [review]
This looks good to me. I am always a be wary when converting those tests that we could introduce an error. Did you try to introduce an error (like use the wrong) version just to make sure that the test fails as expected?

File: sw/device/silicon_creator/rom/e2e/boot_policy_newer/BUILD (hunk 1 above)
Comment (cfrantz) [reply sw/device/silicon_creator/rom/e2e/boot_policy_newer/BUILD]: Done.

File: sw/device/silicon_creator/rom/e2e/boot_policy_rollback/BUILD (hunk 2 above)
Comment (cfrantz) [reply sw/device/silicon_creator/rom/e2e/boot_policy_rollback/BUILD]: Done

**Review by moidx — APPROVED** [review]

**Pull request merged**